src/CharTable.jl

```diff
@@ -12,46 +14,44 @@ abstract type AbstractGenericCharacter{T} end
 # to fix possible interplay issues caused by using multiple tables.
 # See for example A1/GL2.jl
 struct CharTable{T} <: Table{T}
-	order::T  # Order of the associated group
+	order::UPoly  # Order of the associated group
 	classinfo::Vector{<:Any}  # Info about class types
-	classlength::Vector{T}  # Order of the classes in each type
+	classlength::Vector{UPoly}  # Order of the classes in each type
 	classsums::Vector{Function}  # Functions to sum a Cyclotomic over all classes in a type
 	classparamindex::Vector{Int64}  # Indices of the class parameters
 	charparamindex::Vector{Int64}  # Indices of the character parameters
-	classparams::Vector{Parameters{T}}  # Info about the parameters of each class type
-	congruence::Union{Tuple{T, T}, Nothing}  # Congruence of the main parameter q (of T). q is congruent to congruence[1] mod congruence[2].
-	modulusring::PolyRing  # Ring of polynomials of type T used in table (also ring of modulus of Cyclotomics)
-	argumentring::Generic.UniversalPolyRing{Generic.FracFieldElem{T}}  # Ring of argument of the Cyclotomics in table
+	classparams::Vector{Parameters}  # Info about the parameters of each class type
+	congruence::Union{Tuple{QQFieldElem, QQFieldElem}, Nothing}  # Congruence of the main parameter q (of T). q is congruent to congruence[1] mod congruence[2].
+	ring::GenericCycloRing  # Ring of the Cyclotomics in table  # TODO rename to parent?
```

Comment:
This ring is not really the parent. Maybe `base_ring` would fit but that is also a bit of a stretch? `coefficient_ring`?  well, or just leave it as ring. 

So like this, including some typo fixing
```suggestion
	ring::GenericCycloRing  # parent ring of the cyclotomics in this table
```